---
layout: post
title:  "ENCALS Oxford 2018 day two"
date:   2018-06-21
categories: ALS
---

For day one, see [here]({{ site.baseurl }}{% post_url 2018-06-20-ENCALS-Day-one-report %}).
For day three, see [here]({{ site.baseurl }}{% post_url 2018-06-22-ENCALS-Day-three-report %}).

---

### Summary of first day

I confess I was a bit nonplussed by the yesterday's first session, in which it was suggested that hyperexcitable neurons are important (but doubt was cast on how specific to ALS this is), that exercise & athleticism may be associated (but evidence wasn't presented except the [Italian footballers](https://www.ncbi.nlm.nih.gov/pubmed/19267274); surely there are other footballers around to test this in?), that electric shocks or exposure to high frequency electric/magnetic fields may be causal (though there's some doubt about confounding).  In the second session we heard that telomere length is associated with increased risk, an opposite effect to that [found in mice](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4789589/).  

Of course, any of these could be true, but surely they can't *all* be true.  (But given the rarity of disease, they could all be noise.)  I'm left not quite knowing what to believe.

It's also clearly not known how currently-used drugs work, and anyway they don't seem to work very well.  

So what I think this says is the field is still searching for the real causes.

Not surprisingly I liked the genetic talks yesterday better (as well as the talk from Jill Meier on spread of neuron impairment along the brain [connectome]((http://www.dutchconnectomelab.nl))).  That's maybe not surprising as I'm a geneticist. The first talk of this session said we need large GWAS. There are some clearly real genetic signals around (particularly in the versions of ALS that run in families).  Genotypes are easier than most things to reason about causally, because the (adult) outcome can't generally affect the genotype.  (Although there was an observation of somatic C9orf72 variation yesterday, which was also interesting).  The high estimated heritability means that a lot of the causal effects must be genetic (although the first speaker yesterday kind of contradicted this).  Some of the specific signals are a bit worrying - it's hard to get too excited, for example, about the odds ratio of 32 for the rare variant burden test in *[KIF5A](https://www.omim.org/entry/602821)*.  (This resembles the Italian footballers in that it is based on a small number of observations of mutants among many many individuals observed.  Even if there's no bias it must suffer from [Winner's curse](https://en.wikipedia.org/wiki/Winner%27s_curse), and if it was that strong, surely it would have replicated?)  The GWAS signal in the same gene is clearly real, though, so maybe it is true.

On to day two's talks:

---

### Session 3

*James Shorter* (U Penn)
"Reversing aberrant phase transitions connected to ALS"
About [TDP-43](https://en.wikipedia.org/wiki/TARDBP) and [FUS](https://en.wikipedia.org/wiki/FUS_(gene)), which form cytoplasmic 'inclusions' in the degenerating neurons of ALS and FTD patients.  TDP-43 pathology is very common in ALS (93% of patients), while FUS pathology is rarer, maybe 1%, a bit more seen in FTD, which also sees TDP-43 and [TAU](http://www.uniprot.org/uniprot/P10636) pathologies.  Both TDP-43 and FUS are RNA-binding proteins (RBPs) with 'prion-like' domains.  They shuttle in and out of the nucleus, performing RNA transport, and function in splicing, tanscription & RNA processing.  They help regulate thousands of human genes.  "[Prion-like domains](http://www.ncbi.nlm.nih.gov/pubmed/28389532)" (PrLDs) are the distinctive low-complexity domains enriched in Q,N,S,Y and G, which enable RBPs to form self-templating fibrils.  They are a _type_ of low-complexity domain.  A prion is an infectious protein that can exist in at least two forms, one of which is self-templating.  Template proteins alter native proteins into self-templating state.  c.f. Mad cow disease.  Prion proteins are found in yeast (SUP35, URE2, RNQ1).  It is the domain that's important, e.g. if domain is deleted, SUP35 loses ability to template.  Moreover inserting the domain into another protein makes it prion-like.  Also, can scramble the amino acid sequence of the domain: it still acts like a prion.  So it's the composition, not the sequence that's important.  Developed algorithms to find PrLDs in genomes.  In humans, find 240 proteins with prion-like domains.  ~30% are RNA binding proteins, another ~30% ar DNA-binding, and ~75% have an nuclear localisation sequence - oftern nuclear proteins.  Ranked list of proteins: FUS, TDP-43 in there, wondered about other genes on this list. Lots of refs here e.g. [Harrison & Shorter 2017](http://www.ncbi.nlm.nih.gov/pubmed/28389532).  Indeed several of these genes are now linked to neurodegenerative disease.  E.g. TIA1 connected to rare forms of ALS.  It seems to be expansions within these prion-like domains.  Example is TDP-43, where almost all the disease mutations cluster in the PrLD.  So why do humans keep these domains?  It turns out the domain is important for the function of these genes.  Now showing experiment observing phase transition of protein into pathological fibrillar agregate.  (e.g. Murakami et al Neuron, [March et al Brain Res 2016](http://www.ncbi.nlm.nih.gov/pubmed/26996412)).  In liquid phase, these proteins clump and eventually convert into a gel-like phase, an 'aberrant phase transition'.  Are these aberrant phase transitions actually neurotoic?  In TDP-43, tag TDP-43 with Cry2 light-responsive protein, now can shine blue light on cell and induce aberrant phase transition and aks whether it's toxic to cultured neurons.  Timelapse photos of TDP-43 in neurons under light and no light.  It behaves different.  Aberrant TDP-43 spreads under blue light and is damaging to cells.  Can we find agents that reverse the aberrant phase transitions?  Attractive strategy because could reverse toxicity.  Using Karyopherin-beta2 (Kapb2, or transportin), a nuclear-import receptor for FUS, TAF15, EWSR1, and many of the proteins on their list of prion-like proteins.  Trafficking proteins through nucleur pore (which is itself a complex viscous gel). Ran inside nucleus seperates KapB2 from cargo.  Now Guo et al Cell 2018 <>.  Found Kapb2 rapidly disassemble FUS fibrils.  Then tried KapB2-WWAA (modified KapB2) that can't disassemble FUS.  Adding ran suggested this dissassembly activity is likely localised to the cytoplasm, not the nucleus.  What about other FUS binding proteins?  E.g. HDA1, FUS antibodies, did not have these properties, so simply binding is not enough.  So what's going on?  Seems that KapB2 engages PY-NLS on FUs fibrils (liek a FUS antibody), but also makes secondary contact with the PrLD of FUS, and this leads to disaggregation leading to soluble KapB2-FUS complex.  Ran-GTP completes the disaggregation.  Ran GTP then lets KapB2 separate KapB2 from FUS.  What about other RNA binding proteins?  KapB2 works for TAF15 and another PrLD protein fibrils.  Extends to others. But annoyingly, not some mutant forms - FUS-R495X and FUS-P525L.  (What about TDP-43?  He doesn't mention that here).  Oh, here w go: Importin alpha and KapB1 dissamble TDP-43 (as does something called A503.).  Now looking at macroscopic images.  FUS apontaneously self-assembles into macroscopic hydrogel stages.  What does Kap-B2 do to this?  Sure enough, KapB2 completely disrupts structure.  Pretty striking.  Now real-time video of FUS liquid droplets being dissolved.  Now in yeast cells.  FUS is toxic to yeast cells.  Switch KapB2 on after FUS has accumulated.  Eliminated FUS in cytoplasm.  We would like to be able to do this in ALS patients.  Now showing reverse of neuron degeneration in drosophila (fruit flies).  Conclusions.  Some sort of nucleur import defect leads to FUS localising into cytoplasm, leading to solid stage aggregates.  Think KapB2 can reverse all of these stages bringing it back to normal state.  A parallel system (Importin alpha + KapB2) also does it for TDP-43.  (Would also be good to look for small molecules / drugs that increase KapB2 expression.)  Future: re-engineering KapB2 to recognise mutant FUS forms, preliminary data says this may be working.  (This talk was nice.)  Questioner asks about inside the nucleus, where this system is not going to work because of RAN GTP levels.  Speaker says yes seomtimes you see aggregates in the nucleus and he'd like to know if there's a parallel mechanism there.

*Raphael Munoz-Nuez* (Paris ICM) - studiying interaction between TDP-43 and SQSTM1/P62
Most common ALS genes discovered are involved in protein clearance; another group due to RNA metabolism as in previous talk.  TDP-43 has a nucleic acid binding domain, PrLD leads to phosphorylated and ubiquitinated TDP-43 aggreagates.  (Line et al 2013, Neumann et al 2006). Now sequestosome 1/p62.  Looking for protein-protein or protein/RNA level interactions between these two proteins.  Work on zebrafish, transient knockdown models.  Not really listening to this.

*Piettro Fratta* (UCL)
TDP-43 central to ALS pathogenesis.  TDP-43 mutations clearly established as causing disease.  We know a lot about it.  (c.f. [Lagier-Tourenne et al](https://www.ncbi.nlm.nih.gov/pubmed/20400460).)  TDP-43 is dosage-sensitive.  Even subtle alterations in protein levels induce splicing changes.  About experiments trying to get normal expression (I think).  I'm not following this.  A conlusion is TDP-43 gain of function mutations produce novel splicing events "skiptic exons".

*Jolien Seyaert* (Leuven)
FUS inclusions are found in ALS and FTD pateients.  Aim: characterise FUS toxiticy in Drosophila model.  Developed 3 fly lines in which human FUS gene + 2 mutant forms were inserted.  Expressed in central nervous system using a GAL4 expression system.  Pictures of mice that are maldeveloped (poor wings and skin).  Try to udnerstand this.  Study particular neurons called CCAP neurons. A subset of these secrete Bursicon, is this goind wrong?  Seems to be less Bursicon being expressed by these neurons.  CCAP neuron loss.  Can be rescued by preventing apoptosis.  Table of human families from [Kwiatkowski et al 2009, Science](https://www.ncbi.nlm.nih.gov/pubmed/19251627), point is that there's a wide range of age of onset and also disease progression within same mutant carriers (and within families).  Now talking abotu 'exportin 1'.  Exportin 1 knockdown counteracts the formation of FUS inclusions.  (I thikn FUS inclusions = the FUs accumulations described in earlier talks, but not sure.)  Hypothesis: exportin binds to FUS as it leaves nuclear complex, and sequesters it into 'stress granules', where it aggregates.  No exportin => less aggregation => protective.  Says NUP154 and Exportin 1 are potential suppressors of FUS neurotoxicity.

*Somebody* (somewhere) (I thought this was Ziquang Lin, but she spoke on day 3 on a different topic.  So this is somebody else).
'SOD1 prions transmit ALS to hemizygous hSOD1^D90A transgenic mice'
SOD1 exists in all cells i human body, mutations are known to cause ALS.  'Seeding principle', modified from [Jucker et al 2013](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3963807/).  Natural initial phase of fibril formation is slow.  Injecting seed SOD1 fibrils leads to much quicker growth of aggregates and death (of mice).  Are SOD1 proteins from mutant lines seeds for this?  Try by injecting them, using two strains ('A' and 'B') and control strain.  1ul of seed (~5ng hSOD1).  Strain A and B mice, seeded much quicker death.  But control strain and not-inculated strain didn't show this.  (So I think she is saying the mutant SOD1 is not seeding.)  (I forgot to say: poor mice.)  Now another SDO1 mutation discovered in a patient in 1995.  (Andersen PM et al, Brain 1997)[https://www.ncbi.nlm.nih.gov/pubmed/9365366].  This talk had lots of similar-looking plots, I got a bit lost.  Conclusion is that 'strain A'-like truncated humaised SOD1 did induce a transmissible aggregation causing disease and death.

---

### Coffee break

Over the break I was looking at posters, some of them are:

- a poster about assessment of microsatellite repeats using whole-genome sequencing data from [Project MiNE](https://www.projectmine.com).  (They use [HipSTR](https://github.com/tfwillems/HipSTR) and other existing software).

- a poster about [Masitinib](https://en.wikipedia.org/wiki/Masitinib).  Masitinib has undergone an initial phase III trial in which it appeared successfully therapeutic, with modest effects on ALS.  It is now undergoing a [second phase III clinical trial](https://clinicaltrials.gov/ct2/show/NCT03127267) that is recruiting in 2018.  (See also [this](https://alsnewstoday.com/2017/05/19/masitinib-slows-als-progression-effective-treatment-trial-data-shows-says-ab-science/), which I think is about an earlier trial.

---

### Session 4

*Olaf Ansorge* (Oxford) - Neuropathological heterogneity across ALS.
Starts with outline, with 6 themes: What do we talk about when we talk about ALS?  (Nosology)[https://en.wikipedia.org/wiki/Nosology].  Phenotypic extremes.  Does 'incidental ALS exist?  Selective vulernability.  Genotype and neuropathological phenotype: powerful allies.  Finally a clinical vignette.  1: ALS is primarily a clinical diagnosis - upper and lower motor neuron signs and symptoms.  Clinician infers _amyotrophy_ (affecting lower motor neurons (LMN)) and _lateral sclerosis_ (UMN).  Anatomy of UMNs connected to LMNS in brain and spine.  Pictures of two real spines from a healthy and an ALS patient.  Spine is visibly altered with neurogenic atrophy (the amyotrophic component).  Now cross-sectino of spinal chord, severe degeneration where spinal column material becomes soft.  So initial signs are clinical.  Can then further refine by molecular pathology.  Extremes of phenotype: what are the boundaries of ALS?  Now presymptomatic neuropathology.  Plot (I think it's a cartoon) showing protein aggregation of TDP-43 prior to clinical diagnosis.  Does one ever see TDP pathology in individuals not suffering from ALS?  He thinks no, because it's never seen in individuals in their brain bank.  This differs from Alzhemiers and Parkinsons disease which shows a linear trend.  In ALS postulates that onset of protein aggregation is a fairly rapid "catastrophic" event.  That's pretty interesting.  (Notes there may be some reports of this from Asian studies.)  "Selective vulnerability".  95% of ALS is defined by mislocalisation of TDP-43, aggregating into fibrils.  A classification scheme was established for FTD field, types A, B and C, with different types of aggregation linked to different clinical presentations.  Sporadic cases seem to be in 'type B', C9 cases seem to be mixture of type A and B.  What that means is not entirely clear.  There is a striking oligodendrogliopathy in ALS.  [Takeuchi et al 2016](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4918136/).  Seen in grey matter, rarely in white matter.  What does this mean for propagation?  (Not all oligodendrocytes are created equal, some are 'satellites', some are 'interfascicular', they do different things, understanding this -opathy in these would be valuable).  Now intra-individual versus inter-individual TDP-43 'strain' diversity.  Is there diversity within individuals?  E.g. can't find specific C-terminal fragments in spinal chord, suggesting it is specific to parts of the nervous system, although cautions this could be an artifact.  Now talking about major effector proteins.  SOD1, TDP-43, FUS.  Genes and pathology define beginning and end of pathogenic pathway.  TDP-43 in neurons and oligodendroglia.  SOD1 in neurons and maybe astroglia.  FUS in neurons and oligodendroglia.  TDP-43 seems to be the pathological effector of mutations in a range of genes.  Lastly clinical vignette: a man with clinical diagnosis of FTD.  Sequenced, found het for TDP-43 mutation (c.859G>A, p.(Gly287Ser).)  Children at 50% risk of inheriting this mutation.  This man came for autopsy and did full research autopsy.  Conclusion is that this is full-blown alzheimers pathology, no evidence of TDP-43 dysfunction, this is a alzheimers FTD mimic.  Brain sequencing confirmed this.  Notes: genetic services have an impossible task if they are required to study in detail the evidence in the literature before issuing a report.  Ends with picture of the [Oxford Radcliffe Observatory](https://www.gtc.ox.ac.uk/about-gtc/history-and-architecture/the-radcliffe-observatory.html).

*Chris Henstridge* (Edinburgh)
Synapse loss in the prefontal cortex.  Evidence that ALS and FTD lie on the same spectrum: genetic evidence; pathological evidence (half of FTD patients also present with ALS) - but also clinical evidence re: symptoms. a 3rd of ALS patients also have cognitive decline that is similar to but not diagnosed as FTD.  Could similar brain changes underlie this?  Focus on [synapse](https://en.wikipedia.org/wiki/Synapse) loss, which is a shared mechanism in neurodenegeraticve disease.  Early synapse loss is seen early: before neuron loss, in PD, AD, ALS, FTD.  Data is from postmortem tissue analysis.  20 ALS and 5 control patients.  Tissue preserved in many formats for different research approaches, enabling tissue electron microscopy.  observe ~100 synapses per case.  Decrease in ALS patients is statistically significant.  Shows video of [array tomography](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2080672/).  Cut cortex columns, fresh pieces of brain are embedded in resin, can cut ribbons 70nm in thickness.  Can do conventional approaches on these ribbons and then reassemble into 3d map.  prefontal cortex (29 ALS and 14 controls.  ~36,000 synapses per case).  Difference is still statistically significant (but no more so than before, there's more data here so I guess this indicates the observed effect is smaller).  Of 29 ALS cases, 2 had been cognitively screened pre-mortem.  16 unimpaired and 7 impaired meaning they have a subtle cognitive change not as strong as FTD.  Impaired patients has the lowest synaptic counts.  Conclusion: synapse loss is associated with cognitive change.  So what's driving that?  Found no association between alzheimers-like pathology and synapse density.  pTDP-43 does associate weakly with lower synapse density.  pTDP-43 aggregations accumulate at the synapse in ALS.  Picture show large clumps but also a lot of small clumps which localise with synapses.  But, just because they're there, are they causing anything?  Trying to get at this at the moment, need to use mouse models.  Single Q331K mutation knock-in.  Found 'frontal dependent behavioural change'.  [TDP-32 gains function due to perturped autoregulation... ](https://www.ncbi.nlm.nih.gov/pubmed/29556029).  Can't say what mechanism of synaptic loss is, but can say that that single mutation is leading to synaptic loss.  Also [this](https://www.ncbi.nlm.nih.gov/pubmed/28669544).  Microglia-dependent synapse loss?  In TDP-43 KO in microglia results in fewer cortical synapses, more synaptic material engulfed by microglia.  Summarises and describes ongoing studies on this.

*Noemi Gatto* (Sheffield)
About misfolded SOD1 in [astrocytes](https://en.wikipedia.org/wiki/Astrocyte).  Diagram of pathogenic mechanisms in SOD1 ALS.  Aim to determine role of wild type SOD1 in sporadic ALS cases, using iNPCs (induced neural progentor cells).  Slide about making these.  Is SOD1 in the nucleus?  Staining indicates it is.  Found higher levels of nuclear misfolded SOD1 in sALS (but sample size is tiny - 3 in total I think).  shRNA ([short hairpin RNA](https://en.wikipedia.org/wiki/Short_hairpin_RNA))for SOD1 successfully reduces the level of SOD1.  Also observe higher level in C9 patients.  (Not clear to me if these are all different from the previous ones).  SOD1 mutation patients don't have a higher level of nucleur SOD1.  I don't think any of these sample sizes are large.  Nw, does CRM1 (which I think is Exportin 1, see [here](http://www.uniprot.org/uniprot/O14980)) decrease?  Preliminary data suggest yes there's a decrease.  So these data indicate SOD1 detectable in nucleus of astrocytes and reveals link between possibile unknown function of SOD1 and sALS pathophysiology.

*Matthew Nolan* (Oxford)
About selective vulnerability in ALS.  Says several cells (not just neurons) are involved, and proteinopathies seem to be involved in all of these.  Picture of (real I think) primary motor cortex and brain.  Showing parts controlling face, hand, leg. Huh.  Aims to characterise pathology of primary motor cortex across spectrum of ALS, and compare this across molecularly-defined subtypes.  Use a variety of markers to assess selective vulnerability across specific cell types.  A problem is that these markers don't work well in long-fixed material found in most biobanks.  Using short-fixed material this works.  Study has 19 controls, 43, 18, 11, 9, 5 sporadic, C9orf72, SOD1, FUS, and other gene-variant pateints of ALS.  Severity of pTDP-43 deposition correlated with extent of microglial activation.  Now looking at UMN versus LMN burden of [CD68](https://en.wikipedia.org/wiki/CD68) (which I reckon is one of their chosen markers).  I got distracted here.

*Roisin McKackin* (Trinity College, Dublin)
ALS as a network disorder.  Measuring network change.  Pros and cons of electroencephalography (EG).  Pros: 1: gives a direct measure of neuronal function; 2: dysfunction leads to cell death and can measure change before cell loss; 3: the cost is tens of thousands rather than millions (for [MEG](https://en.wikipedia.org/wiki/Magnetoencephalography), [MRI](https://en.wikipedia.org/wiki/Magnetic_resonance_imaging), [fMRI](https://en.wikipedia.org/wiki/Functional_magnetic_resonance_imaging), PET).  Cons: spatial resolution is not as high.  However 'source localisation' techniques allow to improve resolution, c.f.  [Muthuramen et al 2014](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0091441).  A network of electrodes allows to transfer EG to a high-resolution map.  Example: 'mismatch negativity', EEG giving poor resolution.  Investigated using source localisation methods.  Dipole fitting, LCMV, eLORETA. 58 ALS patients, 7 had C9, 12 with family history (+1 with family history of FTD).  Results: diploe fitting, gives significatn effects in inferior and superior [fronal gyri](https://en.wikipedia.org/wiki/Frontal_gyrus).  Now using eLORETA which has relatively low spatial resolution.  Nevertheless gave reliable info and proceeded with this.  Couldn't detect a significant group-level difference.  Now result using LCMV which uses a beam-forming approach to localisation.  This seemed to detect better (I think).  Using 'empirical bayesian significance testing with a FDR of 10%'!  Yay!  Colour-word interference test in 27 patients suggests localised activity does reflect impaired cognitive flexibility.  Hmm.

--- 

### Lunch

Lunch. Sandwiches, crisps, water, banana.  Look at posters.

---
*Christine Holt* (Cambridge)
Axonal mRNA biology: implications for axonal maintenance.  Working on embryonic visual system - retinal ganglions, that form the main axons that form the optic nerve.  In tadpoles.  Over the years have discovered queues during growth that guide growth and branching of the axon.  Also survival queues that are important to maintain the axon.  Central Dogma: DNA -> RNA -> protein.  In axons, proteins get shipped out along the axons but this could be a problem if they are long.  It is quite slow: might take a couple of days along a long axon: this isn't much good if you need fast response.  Cells have come up with another mechanism: transport RNA and translate it in far reaches of cell.  It is mostly repressed during transport. Then a signal comes in that activates protein synthesis.  Cites Campbell and Holt 2001; Ming et al 2002; Brittis et al 2002; Wu & Jaffrey 2005, Leung et al 2006, Yao et al 2006.  Evidence: can artificially guide axons growth.  Adding a protein inhibitor, axons continued to grow.  Also true if axons was cut.  "Netrin gradient elicits asymmetric 'near-side' beta-actin synthesis."  RNA trafficked to axons via RNA-binding proteins (RNPs, e.g. ZBP1, FRX, FUS).  Cool video of beta-actin RNA moving along axon during axon growth, also see mitochondrial movement, a dynamic process.  EB1 (indicator of microtubules) also moving.  Can look live at the translation of an mRNA in vivo.  Video of simultaneous image of rna and translation.  That is very very cool.  Laser capture of growth cones: 1000s of mRNAs in frog and mouse axons and growth cones.  Spanning many categories of function, but really wanted to know what is actually being translated in axons _in vivo_.  [Shigeoka et al Cell 2016](https://www.cell.com/cell/abstract/S0092-8674(16)30580-3).  Use 'Axon-TRAP-RiboTag'.  Immunoprecipitation of ribosome-mRNAs in retinal axons.  Get different stages during growth.  "Translatome is developmentally regulated".  Shows gene ontology analysis from the [same paper]((https://www.cell.com/cell/abstract/S0092-8674(16)30580-3).  Now talking about Lamin B2.  Synthesised in axons in response to growth stimulation.  Axon survival requiers axonal translation (of LB2).  LB2 lacking nuclear localisation sequence rescues axon degeneration.  LB2 colocalises with mitochondria and knockdown leads to mitochondria dysfunction.  Now is there a link from FUS and axonal translation?  Shows again gelling of FUS mutant proteins (as in a previous talk, see above).  FUS mutants decrease protein synthesis in axons.

*Wenting Guo* (Leuven) - I didn't listen to this.

*Axel Freischmidt* (Ulm) - I didn't listen to this.

*Laura Fumagalli* (Leuven)
C9orf72 expansions cause axonal transport defects in iPSC-derived neurons.  Didn't listen to this either but there's a cool video of movement of mitochondria along axons, impaired in C9orf72 mutant iPSC-derived axons.  This is in real time but sped up about 100X or so.

*Jik Nijssen* (Stockholm) - About 'Axon-seq', I guess about [this](https://www.biorxiv.org/content/early/2018/05/14/321596).
Axon is the longest cell in body.  Muscle denervation and axon retraction occur first in ALS.  Motor neuron somas are lost later in disease.  Used microfluidic system to isolate motor axons.  Mouse derived motor neurones.  Use gradient of growth factors to induce axon growth through microfluidic channels.  Pretty cool!  Stained with MAP2 and TAU to ensure not getting dendrites, but getting motor axons.  Now axonal material can be harvested separately.  Can isolate axons from rest of cell.  Did RNA seq applied specifically to motor axon compartment.  Shows some QC.  Axonal mRNA from all thousands of axons amounts is about same as mRNA from single isolated motor neuron.  So not very much.  But get signal across axons.  Multiple differentially expresse dgenes (n=771 axon-enriched).  n>200 entirely axon-excluded transcripts.  Shows 25 highest-enriched-in-axons genes.  Enrichment analysis too.  To validate this considered published studies of primary motor axons, primary drg axons (Saal et al 2014, Gumy et al 2011, Minis et al 2014), core signature of 1750 genes expressed in all of these datasets.  In data, surprised to see a number of transcription factors (known for function in nucleus) turning up.  But seems like they may have function otuside nucleus.  Example: [YBX1](https://www.uniprot.org/uniprot/P67809) showed very high expression in axons and enriched in axons.  It is involved in RNA transport and in splicing - in fact in every step.  Now consider axons from SOD1 mutant mouse line.  Found differential expressed genes.  ALS-linked genes: NEK1, MGRN1, NRP1.  Perspectives: want to transition system to humans (using human iPS lines).  Also want to make it more 'in-vivo-like' by introducing muscle cells into the system, hope will help stabilise their transcriptome.

*Laura Ferraiuolo* (SITraN, Sheffield)
About mRNAs secreted by C9or72 patient-derived astrocytes.  iNPC direct conversion as described this in a talk morning.  Astrocytes from C9 patients induce motor neuron death.  (Meyer et al PNAS 2014).  Now about extracellular vesicles (EVs).  Misfolded proteins (a-synuclein, SOD1, something else) found in EVs.  ref [Aoki Y et al, Brain 2017](https://www.ncbi.nlm.nih.gov/pubmed/28334866).  Decreased EV biogenesis protein transcripts in C9 astrocytes.  Shows funky microscope detecting by shining lasers.  Or something.  EVs express CD63 but not CD9.  Most RNA content is mRNA.  Quantified this using GeneChip miRNA array 4.0.  A hundred or so upregulated, 70 or so downregulated.  (This seems like more data is needed).  'Axon guidance' is the most enriched pathway though.  Huh.  Focus on miR-146a for a bt

*Caitia Gomes* (Lisbon)
About astrocytes from ALS patient fibroblasts.  Only two drugs for ALS - riluzole and edaravone only slightly reduce disease (if at all).  There is no cure.  Astrocytes play a role in ALS through interaction with motor neurons, expressing neurotoxins.  Some evidence of miRNA dysregulation in astrocytes in ALS patients.  Goal to understand the mechanisms of this.  Mouse SOD1 model.  Focus on miR-146a.  Looking at neurotoxic effect.  I'm not listening to this.

---
### Session 6

*Ana Candalija* (Oxford)
Transciptomic analysis of iPSC-derived motor neurons from C9orf72 ALS/FTD patients.
Jings.  This talk is more cell visualisation.  It's all good stuff (presumably) but I'm running out of concentration :(.  They are gonna use C9-corrected lines using CRIPR/CAS9 editing to work around natural experimental variability.  Anyhoo.  they find differentially expressed genes in C9 versus edited lines.  SYT11 is one of these.  (GB: It's of course good to look at differentially expressed genes.  But just because genes are differentially expressed does not make them causal.  They could be differentially expressed because they are influenced by other molecules that are in the causal pathway, or because the causal pathway induces large global hanges to the cell.  In fact, given the complexity of how transcription works, it's not impossible that minor hanges in expressino of protein 1 (a causal protein) effect major changes in expression on many other proteins.  Unlikely studying these proteins will lead to therapies.  Thus maybe it's better to look (for example) for transcription factors that influence all of the differentially expressed genes?  Anyway that's not what these talks have done).

*Hortense de Calbiac* (Paris ICM)
Synergistic mechanisms of C9orf72 gain and loss of function.  ALS has strong genetic composition, more than 20 genes involved.  Diagram of autophagy pathway.  E.g. Lee JK et al biophysica (I think). Pathogenicity of C9orf72 mutation. Summary of C9orf92 repeat expansion, found by GWAS and finemapped.  So zebrafish model.  PLot swimming path of zebrafish in petri dish after touch (touch-evoked escape response).  Video of this.  Cool!  They move!  Quantify this in C9orf72 knockdown and 'mismach' (dunno what this is).  C9 loss-of-function disrupts poly(GP) clearance. (I think poly(GP) is a marker of the C9 repeat expansion, c.f. [this](http://stm.sciencemag.org/content/9/383/eaai7866)).  (Again, I am losing this talk.  Like many of these lab-based experiment talks, it consists of slide after slide of bar plots that show some kind of effect.  The axes are labelled differently.  They mean similar and/or different things.  It's impossible to understand unless you already know what the experiment is. For example - suppose I don't know what P62 is (I don't).  How should I understand 


*Matthew Wood* (Oxford, also spin-out companies that he will mention).
Starts with slide of human conditions with known molecular basis from [OMIM](https://www.omim.org).  Many thousands, while only ~500 have therapies.  Prospects for genetic medicines.  Genome-based theraputic technology have large potential.  Significant recent profress.  Realising this wil depend on overcoming intracellular drug delivery.  Ok, nucleic acid-based drugs exploit a variety of molecular mechanisms.  gene expression, gene splicing, gene silencing, gene editing.  Most companies using first-generation technologies - they are working, but not well enough.  Example.  Eteplirsen for DMD (muscular dystrophy), approved 2016 by FDA.  Controversial decision as it has very little effect (but it's safe).  Nusinersen for spinal muscular atrophy (SMA), approved 2016 FDA.  The efficacy in this case is actually pretty good, probably this is an exception.  And Huntingdon's disease a number of companies developing oligonucleotide drugs.  But, speaker's perspective is that clinical benefit for most of the will be modest - at best.  Reason: intracellular delivery is the major barrier.  The drugs don't get into cells readily but stay outside.  Need to get them into the cell and into the compartment it needs be active (e.g. nucleus for oligonucleotides).  This is the major scientific barrier to effective genetic medicines.  Now about developing next-gen oligonucleotides.  Two areas for development are the backbone chemistry of the drug (will give an example of 'sterochemisty'), and delivery, he will talk about peptide delivery and nanotechnology.  Now talking about 'WAVE platform" exploiting stereochemistry.  By WAVE Life sciences.  Oligonucleotides can occur in left- and right-handed form.  This wasn't previously appreciated.  In a 20 nt oligont, each pair of consecutive nts be linked left or right handed, giving 2^19 variants.  Does that matter?  Well it might. Now have technology to try to generate a sterochemically optimised drug.  Have worked through this for muscular dystrophy.  E.g. Exon 51: increased Dystrophin restoration, stereochemically optimised drug much better.  Clinical trials in last quarter of 2017.  WAVE also developing drugs for C9orf72 which are sterooptimised. Now talking about peptide delivery technology.  Developed with Gait and MRC LMB.  New spin-out company PepGEN.  In Muscular Dystrophy, benchmark against FDA-approved drug.  Peptide-delivered drug is significantly more active.  Think it is 100 to 1000 times more effective.  In mice can reach ~80% of normal levels of dystrophin, generating almost complete physiological restoration, whereas standard drug gives ~1%.  Have tried this in SMA.  Nusinersen drug modulates splicing to generate full-length SMN2 mRNA transcript.  But is delivered [intrathecally](https://en.wikipedia.org/wiki/Intrathecal_administration) (i.e. into spinal canal).  With peptides, in mice, again can generate increased levels of delivery through intravenous delivery.  Restores life span and physiology.  This is going into clinical trial.  Think can do something similar in Hungtingdon's.  (Hammond et al 2016.)  PepGen is developing peptide platform tecnology for nucleic acid drug delivery.  Now talking about exosome based nanotechnology.  Some kind of natural nanoparticle (extracellular vesicle).  I dunno what.  COuld we use these for delivery?  We engineer these nanoparticles for this.  Example is exogenous RNA encapsulation, with a rabies virus peptide on surface, to get payload into nervous system. (Mol Ther. 2017).  Deliver double-stranded RNA to mouse cortical neurons, given targetted silencing of BACE-1 gene.  But why was it so effective?  (Heusermann et al JCB 2016).  These natural nanoparticles behave very much like viruses.  They enter as single particles via cell uptake, rapidly transported inside cells.  Shows a model for how they deliver their cargo.  Eventually get degraded by lysosome.  Wow.  Concludes.

A questioner asks if his nanoparticle appraoch can deliver larger molecules - e.g. DNA.  Response is yes, but it's challenging, they are trying to deliver a version of dystrophin, but a cut-down version because it's smaller.

---
*Debate* - "This house believes ALS is a prion-like disease", chaired by [Martin Turner](https://www.ndcn.ox.ac.uk/team/martin-turner).
FOR: *James Shorter* (U Penn)
AGAINST: *Simon Mead* (UCL)

(I'm a bit surprised about this debate.  My reading of the conference is that there's good evidence for agglutination (of several proteins) in ALS.  Witness: the connectome talk, the talk about reversing aberrant phase transitions, the many slides showing aggregating FUS or TBP-42, the one about bioinformatically identified prion-like domains.  And pictures of test tubes with agglutinated FUS.  Isn't this it?  Anyway let's see what the debate says.  We have to vote via a website.)

J.S. defines 'prion-like' (or prionoid) as an inectious protein-like thing that can transmit phenotype wihin an individual and _experimentally_ between individuals.  E.g. alpha-synuclein in Parkinsons.  ALS is not a prion disorder (no transmission between individuals) but there's strong evidence it's prion-like.  Motor neuron loss starts focally in central nervous system (CNS), and spreads during disease progression.  This is compatible with prion-like spread.  Moreover, pathology of ALS also spreads.  [Braak et al Nat. Rev. Neurol 2013](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3943211/).  Evidence from in vitro studies: should be able to spotanrously assemble.  Indeed this has been done for SOD1, TDP-43, not yet done at time of writing paper in 2015 for FUS but probably has now.)  TDP-43 and FUS are both RNA-binding proteins with prion-like domains.  These enable them to assemble into these fibrils.  They get this name because of amino acid similarity to real prions; can insert into genes and make it prion-like; can remove from prion-like genes and get rid of prion behaviour; etc.  To close: to really have definitive evidence, should be able to make recombinant protein in prion conformation and inject (say into mouse).  Mouse should become diseased.  These has been done for PrP (by Ma and Prusiner) and [alpha-synuclein](https://www.michaeljfox.org/understanding-parkinsons/living-with-pd/topic.php?alpha-synuclein)(Lee). Being done for SOD1, TDP-43, and FUS.  E.g. [Ayers et al Acta Neuropathologica 2016](https://www.ncbi.nlm.nih.gov/pubmed/26650262), induced MND.  Not quite there because this is not wild type mouse.  But the weight of evidence is very much toward this.  So yes, ALS is a prion-like disorder.

S.M. Draws attention to wording 'prion-like disease'.  First says he is from the institute of prion disease at UCL.  ALS implicated proteins do show a seeded-like polymerisation mechanism but that's only a small part of the picture.  Mammalian prions are pathogens that invade, evolve, kill host.  "Prion-like" is a poorly defined term.  E.g. prion labs are extremely carefully contained - not so for ALS labs etc!  Shows cartoon of 'seeded polymerisation' mechanism.  How can this explain the diversity of diseases found?  Prion diseases occur in massive epidemics - [Kuru](https://en.wikipedia.org/wiki/Kuru_(disease)) in Papua New Guinea, lost 10% of population, that was transmitted through mortuary feasts.  Mammalian prion diseases - BSE, vCJD, highly contagious. trace amounts of prions can kill.  CJD has strains - 'sporadic' and 'variant' CJD with different clinical phenotypes.   Table of features of ALS versus prion diseases.  In particular evidence of spreading is different than prion diseases.  So a critique of the term 'prion-like'.  If doctor said you had a 'virus-like' disorder you'd want a 2nd opinion.  It's either a virus or it isn't.  Now tackling specific aspects.  *Seeded polymerisation* - well, this could apply to a large proportion of human diseases - c.f. amyloids, Astbury 1935, Jarrett and Lansbury 1993 - pretty much all neurodegenerative disorders - and more.  Almost any protein can adopt amyloid state under right conditions.  'Prion-like' is unhelpful.  What about spreading?  Propagation of pathogen is not propagation of pathology.  Agree evidence that ALS progression is determined by neuroatnatomical pathways.  But most diseases spread, could be gradient of sensitivity, could be disease agent itself.  Spreading is not "prion-like".  Strains of prion diseases are defined by pattern of pathology.  There is heterogeneity in ALS, but these don't correlate with biochemical features of ALS-implicated proteins.  Finally notes there are no implications of the term 'prion-like'.  There are already decisions for prion diseases that give guidance on care, treatment etc.  The data are just not strong enough for ALS. 

This was followed by a discussion.  My sense is that the 'against' argument was not really made sincerely - as he said, the dude is from the institute of prion disease - and he says he will be convinced if the malfolded proteins can be induced in wild type mice.  Most of his argument is about prion strains, and he accepts that the evidence on this may well come over time.  

Despite this it was a victory for the nays.  (However, unfortunately the voting system went wrong, so only 5 people's votes were counted - (including mine :))

See also [day one]({{ site.baseurl }}{% post_url 2018-06-20-ENCALS-Day-one-report %}) or [day three]({{ site.baseurl }}{% post_url 2018-06-22-ENCALS-Day-three-report %}).
