---
layout: post
title:  "ENCALS Oxford 2018: day one"
date:   2018-06-20
categories: ALS
---

I know very little about [the disease known as Amyotrophic Lateral Sclerosis or Motor Neurone Disease](https://en.wikipedia.org/wiki/Amyotrophic_lateral_sclerosis).  I've decided to come to the [ENCALS conference in Oxford](https://www.encals.eu/meetings/encals-meeting-2018-oxford-england/) to find out.  To this end I'm going to do what [I've](https://gavinband.wordpress.com/2010/10/07/gcd-2010-day-1/) [sometimes](https://gavinband.wordpress.com/2011/04/12/emgm-2011/) [done](https://gavinband.wordpress.com/2011/04/12/emgm-2011-day-two/) [before](https://gavinband.wordpress.com/2014/06/08/gem-2014-day-one/): write down everything everyone says and then try to summarise...let's see how it goes.

(See also my notes on [day two]({{ site.baseurl }}{% post_url 2018-06-21-ENCALS-Day-two-report %}) or [day three]({{ site.baseurl }}{% post_url 2018-06-22-ENCALS-Day-three-report %})).

## Wednesday 20th
---
### Session 1

It's pretty packed (not to mention hot) in here - apparently the conference is over-subscribed and I couldn't get into the main lecture hall at first - but later I sneaked in.

*Leonard van der Berg* ([Utrecht](https://ern-euro-nmd.eu/contact/leonard-van-den-berg/)) opens conference with a couple of slides from previous meetings and a quiz on Britishness. (Sample questions: "True of false: Britain owns Australia?", "Have you ever accidentally said 'thank you' to a cash machine?".  etc.)  Much hilarity ensues.

*Evy Reviers* briefly introduces [EUpALS](https://als.eu) - "European organisation for professionals and patients with ALS".  Established 2017.  How can you become a member?  Members are national ALS organisations established in Europe.  Three activity domains: 1: informing and supporting ALS patients, 2: defending ALS patient rights, 3: stimulating research.  New members are charged 1 euro as a symbolic fee.  Members in Iceland France Spain Portugal Italy & elsewhere (but not the UK it looks like).  EUpALS participates in project proposals at European scale.  

Now on to scientific talks:

*Matthew Keirnan* (Sydney) "Roadmaps to therapy in ALS"
Takes tie off because he was told to.  Starts with interesting looking picture which looks like a series of mountain ranges, says this will be explained at end. Talking about ALS in terms of 'transmission'.  He is current editor of <http://jnnp.bmj.com>.  Human motor transmission: machine which provides controlled application of power.
Picture of a crossing in the spinal cord, a spinal nerve, and the lower motor neurons.  Hodgkin and Huxley - I guess this is [this](https://en.wikipedia.org/wiki/Hodgkinâ€“Huxley_model).  They worked on squid axons before the war, but got distracted by the war.  After the war - no squid.  Anyhow they worked out that neurons transmit through sodium channels.  Transmission is like "kangaroo jumping".  Hugh Bostock found ways to study strength/duration of pulses.  How much current do you need to apply to neuron to activate it?  These techniques are now commercial.  Refers to Wainger et al, '[Intrinsic membrane hyperexcitability...](https://doi.org/10.1016/j.celrep.2014.03.019)".  This hyperexcitability seems to increase in ALS patients.  ALS Diagnosis.  We know in MND that axons are dying.  Mentions "Motor Neurone Disease is a clinical diagnosis', [Turner & Talbot](http://pn.bmj.com/content/12/6/396).  Speaker agrees but says technology gives valuable insights.  Interesting slide linking repeat activation of neuron to the clinical aspect of fatigue when using muscles.  Picture of two hands showing classic "[split hand](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3424792/)" presentations of hands (slide says FDI, ADM, APB, I dunno what these are). We are the only species that need fine pincer control; our brains have adapted for this.  Picture of [Lou Gehrig](https://en.wikipedia.org/wiki/Lou_Gehrig) which shows 'split hand'.  Origin of ALS: we know it is a primary neurodegenerative disease, but we haven't worked out how it gets into the [upper motor neurons](https://en.wikipedia.org/wiki/Betz_cell) (UMNs).  Difficulty evaluating UMN dysfunction as no technology to do it.  MRI techniques helpful.  But has been focussing on taking some existing techniques to look at hyperexcitability of UMNs.  Try threshhold tracking of [transcranial magnetic stimulation](https://doi.org/10.1016/j.neuron.2007.06.026) (TMS) in [Riluzole](https://bnf.nice.org.uk/drug/riluzole.html) treated patients.  Similarly in cohorts of ALS patients, some are actually not ALS but TMS can help distinguish them.  ALS - is it all in the genes?  Speaker says, no it can't be.  Onset is a roughly linear relationship with age - "six steps to ALS" (I think he's referring to [this](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4197338/>), which I've read before. I don't understand from this why it can't be genetic).  Cognitive and behavioral symptoms in ALS.  Video of ALS patient showing fasciculation (i.e. [muscle twitches](https://en.wikipedia.org/wiki/Fasciculation)).  Most muscles have twitches, but more fasciculations => greater progress of disease.  Now ALS, physiology & metabolism.  Another thing that's strange: pre-morbid physiology.  Patients tend to develop ALS out of the blue - tend be quite athletic, tend to have normal or low BMI, have reduced coronary artery disease (and so do relatives).  Streesses this is somewhat anecdotal.  Refers to this about [ALS in Italian footballers](http://www.ncbi.nlm.nih.gov/pubmed/15634730).  I've heard this story (about athleticism) but not sure I believed it - interesting to hear it brought up.  If talk to pateients & partners, they report behavioural changes quite a while before clinical symptoms are identified.  Need national population registries (like ENCALS, which they're trying to replicate in Australia as 'PACTALs').  Challenges for ALS/MND in pan-Asian countries - it may have a different genetic makeup.  Now talking about brain structure and ALS transmission.  In summary, there seem to be 'six steps', genes are involved, other factors, and apparently protein spread (c.f. 1960s experiments injecting monkeys with ALS motor neurons).  How do we link transmission and hyperexcitability?  Finally here's the explanation of the first picture: it is energy released by stars when they are dying.  A pulsar is like a hyperexcitable motor neuron!  Er...is it?  Wow.  (It's larger, I should imagine).  Main conclusion: our understanding of ALS is evolving.

In answer to a question, brings up [TDP-43](https://en.wikipedia.org/wiki/TARDBP), a form of which is implicated in frontotemporal dementia and ALS; it also apparently represses HIV transcription.  Another question is about hyperexcitability - it occurs also in lots of other diseases (like stroke), how does he hang that together?  Says he hasn't really quantified that.  Is this higher in magnitude than seen in epilepsy?  Doesn't know.  (See the region [here](http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&position=chr:11072679-11085549).  Maybe it's rs80356717.)

*Boudewijn Sleutjes* (Utrecht)
"Biophysical basis of the acute effects of riluzole and retigabune..."
Starts with hyperexcitability again: symptoms are fasciculations, cramps,  hyperreflexia, spasticity.
Riluzole -> NA+ channel function, inhibits persistant Na+ currents.  It partly normalises hyperexcitability. Reigabine -> K+ channel activator, also reduces MN hyperexcitability. Will focus on 2018 randomised control trial.  It included 18 patients, 4 of which are familial with C9Orf72 mutations).  [Retigabine](https://en.wikipedia.org/wiki/Retigabine) (300mg), Riluzole (100mg), and placebo randomly assigned.  Mixed model analysis applied.  Nerve excitability testing, non-invasive estimation of axonal membrane potential and ion channel activity.  Temperatures were controlled by warming nerves beforehand and throughout tests.  Standard [TROND protocol](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2915523/).  Compared with mathematical neuron model.  Results show no acute effects of riluzole on excitability within 6 hours.  But for retigabine, significant (but perhaps not large) effects were found.  What is the biophysical basis?  Uses simulations, these match pre-dose results very well.  Large input parameter set, varied only one parameter at a time to avoid overfitting.  Post-dose (1.5h) results, best fit had a downward shift of half-activation potential.  Similarly at 6h.  This model suggest hyperpolarizing shift of V~0.5 of slow K^+ channels.  This has some experimental support. Conclusion: excitability is a reliable biomarker, modelling it helps identify mechanism.  Riluzole has no acute effects (maybe time period too short, also they've already been on Riluzole for > 1 year).  Long-term administration may reduce Na^+ influx and Ca^2+-mediated degeneration.

*Jill Meier* (Utrecht) - Connectome-based disease progression in ALS. Can we predict disease progression using MRI?
ALS on a microscale.  pTDP-43 spreading in a "prion-like" manner.  (references Brettschneider et al 2013, Schmidt et al 2016, Braak et al.). Connectome: huge longditudinal dataset, in which patients are scanned.  This works like this: divide the brain into 82 individual regions.  Then use an algorithm to track the connective fibers between regions.  This gives a 'structural brain network'.  (This looks pretty cool).  A link is when there's a strong enough connection observed between brain regions.  We can then measure the link in two different ways - the number of streamlines (like the bandwidth) and the speed of motion, referred to as 'FA'.  (I.e. current and voltage).  In ALS macroscale connectomes, impairment in FA (quality of connection) decreasing over time.  Also it spreads, apparently along the connections (rather than just outwards across whole brain).  Question: can we predict progression with MRI?  First have to answer, 1: what is the progression, can we predict it with a random walk (along the brain network) approach?  39 ALS patients, 4 scans, 5.5 months between scans on average, matched controls.  Refs: [Zalesky A et al 2010 NeuroImage](https://www.ncbi.nlm.nih.gov/pubmed/20600983), find a growing impaired connected component (based on MRI pictures) fron cortex to other parts of brain, identified by comparison with control samples.  (I assume this is a different connected component in each individual).  She's a mathematician but will leave out the formulae. Aw!  "Self-multiplying NOS-biased random walker" (I think NOS = brain network).  I.e. it walks along the observed network.  Main result: compare simulations to real data, straight line fit.  correlation=0.7.  I.e. can predict with r~0.7 the state of progression based on this model.  Also shows differences between 'Brettschneider' stages (maybe [this](http://www.ncbi.nlm.nih.gov/pubmed/23686809).  141 patients with 2 scans, 141 age-gender matched controls.  Something else about prediction between scans.  Conclusion: it works, there's a growing impaired connected component.  Future: clinical features?  Functional data?  (This talk was cool.)

*Alexander Thompson* (Oxford) - CSF chitinae protein performance as ALS biomarkers.
This is a clinically-oriented talk about Chitotriosidase-1 (CHIT1), aldo sCHI3L1, CHI3L2).  Measured using ELISA (which is antibody binding).  Can they be used as predictors?  The significance of these particular proteins is lost on me and I'm not really following this.  Also, while I'm moaning, the measured levels are compared between individuals in a plot with '***' (meaning 'highly statistically significant') and 'NS' (meaning 'not significant').  These are typically not the right way to think about comparisons like this.  For example, in one comparison, all of group two is below the median of group 1 (visually at least), yet this is 'N.S.' and the speaker says there's no difference.  I think that's a flawed way of looking at the data.  Anyway, I'm saying this not because his results are wrong (I'm not really listening) but because I started thinking about it.  Comparison is between ALS, HC, Mim, PS, and AGC patients, but I don't know what the latter 4 are.  Conclusion: chitinases are an emerging CSF biomakrer in ALS.

*Henk-Jan Westeneng* (Utrecth) About imaging in asymptomatic C9orf72 repeat expansion carriers and non-carriers.
Starts with Westeneng et al 2017, but different story in Geevasinga et al Nat Rev Genetics 2014. I missed the slide on # samples but I think it is one extended family.  Use 7 Tesla scanner to generate 3d scan of whole brain.  Investigated all parts of brain - grey matter and white batter, and deeper in brain.    Studied 6 metabolites.  Quality control.  This includes 'Cramer-Rao lower bounds' (huh?).  Bayesian linear mixed model.  Kinship as random effect for family.  Age as covariate.  Weakly informative priors N(0,0.25) on coefficients, T distribution on residual sd.  Computed P-values, adjusted for FDR, and report at 5% FDR.   Results:  GPE (a measure of cell membrane breakdown).  Clear differences, more GPE in C9orf72+ carriers, in all parts of brain.  Also for [PE](http://en.wikipedia.org/wiki/Phosphatidylethanolamine) though maybe less widely.  Also one more, [UDPG](http://en.wikipedia.org/wiki/Uridine_diphosphate_glucose).  Can we use this as a biomarker?  Tried to predict C9orf72 mutation.  Shows prediction intervals for carrier status.  It has nontrivial (but I wouldn't say strong) predictive ability.  Suggests this can therefore be used as a monitor of treatment (if it is in the causal pathway).  Incrase of GPE previously reported in Alzhemiers and Parkinson's disease patients.  Neurodegeneration in pre-symptoamtic patients starts maybe 10 years before clinical diagnosis.

*Greig Joilin* (Sussex)
Non-coding RNA serum biomarkers in ALS.  Schematic of RNA biomarkers (mRNA, ncRNA, tRNA, rRNA, lncRNA, microRNA, piwi-RNA, snoRNA, snRNA.  (Man, there is no way all of these represent distinct biological things.  I reckon they are just a spectrum of the same thing.  Hmm, I was talking to a dude the other day about short noncoding RNAs, he said part of this is that traditional methods couldn't get at short RNAs, which is one reason they're under-studied).  Methods: 24 controls, 17 disease mimics, 24 slow-progression ALS, 24 fast-progressing ALS patients.  RNA-seq on MiSEQ & 75bp paired end, [Oxford Genomics Centre](http://www.well.ox.ac.uk/ogc/about-ogc/) (that's in the WCHG, which is where I work).  Shows breakdown (i.e. pie chart) of types of differentially expressed ncRNAs.  e.g. 12% are miRNAs, 28% are tRNA.  Plot of differentiation goes up to about P=1E-5.  Is this enough to be convinced?  I dunno.  Another plot has the ***'s and 'N.S.'s again. Come on. MIR-A and MIR-B appear correlated with fast/slow progression status.  MIR206 has been reported as up-regulated, but could only detect that in one of their samples, where it has a binary-like pattern.  Possibility that we maybe below detection threshhold.  'Combined signature is 78.5% accurate' in separating healthy and disease.  But this is the same set of samples, right? So it's overfit.  But more samples are coming.

*Susan Peters* (Utrecht)
Electric shock and extremely low-frequency magnetic field exposure and risk of ALS.  ([Euro-MOTOR](http://www.euromotorproject.eu)).  Do 'electrical occupations' play a role in ALS?  Extremely low frequency (ELF) magnetic fields are ubiquitous.  Recent meta-analysis: increased risk, but heterogeneous across ~10 studies (I = 75%).  Stratified studies by studies with full job history and those without.  Get I2=0% and significant increase risk in former set.  Oh!  Interesting.  (But it seems to say P=0.493.  Eh?).  Euro-MOTOR.  1,600 ALS and 3,000 controls.  POpulations-based controls, clinical data + questionaires.  Exposure assessment.  Logistic regression with adjustments for sex, age, alcohol use, and something else.  Results: OR = 1.16 (1.01-1.33) for ELF-MF. ANd OR = 1.23 (1.05-1.43) for electric shocks.  Huh. After adjusting for each other, similar results (but maybe weaker, don't understand this). Compare to earlier findings in Dutch cohort (Koeman et al 2017).  They found clear exposure dependent association for ELF-MF, but speaker did not see this increase with amount of exposure.  Euro_MOTOR suggest both exposures might play a role.  Signals could be driven by something else associated with these types of jobs.  Euro-MOTOR is good, large number of cases clinical confirmed.  But some recall bias (cases think harder about their history.  That is totally relevant here.  Could that explain the effect?  Hard to imagine people can't remember what jobs they've had, but maybe there's a difference in what's reported as involving electricity, especially for low-frequency MF) and controls have higher educational levels.

---
### Session 2

*Michael van Es* (Utrecht) - how many DNA samples is enough?
This talk is intended to be provocative.  And he is Dutch.  So he can be provocative without trying.
How many samples is enough?  Enought for what?  How many is enough to find all ALS genes?  Ok, step back.  Why genetics in the first place?  ALS sometimes runs in families, safe assumption that genes are involved.  Sporadic ALS is also genetic (60% heritable).  COuld possibly be useful for diagnosis, and can aid genetic counselling, but most useful in studying disease pathology.  Gene discoveries get translated into disease models (e.g. mouse models, poor mice).  Do we really need to find _all_ risk factors?  Genetics (slide has SOD1, ALSin, TARDBP, FUS, ANG, VCP, ...) and environment lead to TDP-43 that leads to ALS (prion-like spreading, mitochondrial dysfunction, inflammation, excitotoxicity, axonal transport).  Going to argue it's important to chase the genes.  ONe line is that genetics tells us have different types of ALS.  SOD1, C9orf72, FUS etc.  Will there be a drug applicable to all of these?  Don't know but seems unlikely, so this may mean stratified treatment.  Also caused by different types of genetic variation.  Familial ALS rare => genetic variant also rare, sporadic ALS => multiple risk factors which could be common.  126 "ALS-related" genes reported, but most are not validated.  <alsod.iop.kcl.ac.uk/index.aspx> but many of these are false, and this needs to be cleaned up. Genetics of ALS: in familial ALS have come a long way (SOD1 C9orf72, FUS TDP43).  But most patients are 'sporadic' and we're a long way away from this.  GWAS are probably an effective tool.  Common genetic variants (5% and above), compare between disease cases and controls.  Tells us what a manhattan plot is (<https://en.wikipedia.org/wiki/Manhattan_plot>) and about statistical power (cue slide of bored people sleeping).  Ballpark estimate: 1st study in ALS had 276 ALS ad 271 controls, no convincing signals.  LAtest Van Rheene et al Natugre Genetics 2016 <https://www.nature.com/articles/ng.3622> (12,577 cases, 23,475 controls.)  GWAS have critiques.  One is that they are way too expensive for what they deliver.  But prices dropping, now ~$25 per sample.  Data sharing means can share controls with other studies.  What have we learned from GWAS studies?  Shows heritability by MAF in schizophrenia (SCZ) and ALS.  SCZ mainly driven by more common variants.  For ALS appears to be more rare.  Also shows the 9 'hits'.  MOBP, C9orf72, TBK1, SCFD1, SARM, UNC13A, C21orf2 (that's 7, dunno where the other two are).  The other criticism: risk for each variant is small.  However, even with a small OR on disease can have substantial effect: shows plot of UNC13A variant which has small effect on disease risk but large effect on phenotype / survival time.  Van Eijk et al Neurology (2017).  Now talking about whole genome sequencing.  Gives you SNPs, coding and non-coding genome, structural variants, repeat expansions.  Bar chart of % of gene in different functional classes (1.5% is protein coding).  Project MinE.  Aims to sequence 15,000 ALS cases & 7,500 controls.  Map of participating countries.  Now have 14,000 samples sequenced.  Next data freeze will give more samples and better power, using additional controls obtained from another study.  Describes rare variant burden (RVB) analysis. QQ-plot of RVB analysis in ALS genes works very well (SOD1 at top.  Also lambda = 0.95, why is this?).  What about repeat expansions?  Example: expansion with final P-value=10^-7.  So this is 'exome-wide significant'.  What about the non-coding genome?  Burden test other functional elemetns?  What about follow-up studies (as unlikely to be able to make mouse models)?  Mentions 'organoids' which is one way to go.  Whole-genome sequencing may become standard work on a patient.  Diagnostic analysis for 1 gene = 300 euro.  So cost for a few of these is more than for WGS.  Need to start thinking about how to incorporate this information.  Slide on gene therapy (e.g. antisense study for C9orf72).  Gene delivery, gene silencing, but also need to think about how to show these are effective.  Cross-phenotype studies?  So: what we need to do in genetics is clear, it will be complicated but will get there in the end.  

Marie Ryan (Dublin)
Oligogenic and discordant inheritance: population based genomic study of Irish kindreds carrying C9orf72 repeat expansion (hereafter referred to as C9).  C9 is one of the top 4 causes of ALS.  Known pleiotropic between ALS and other psychiatric disorders.  1022 DNA samples (blood) screened for C9.  269 individuals with familial ALS.  131 individuals in 122 families have next-gen sequencing (Ryan et al NAture Genetics).  Screened for 38 genes considered linked to ALS or FTD.  Results: identified 89 C9 +ve individuals.  reduced survival of C9 carriers.  This talk is going by very fast indeed.  Also 'Oligogenic inheritance' (n=11) but didn't catch this.  Shows one family with C9 and ALS, apparently not co-segregating (different individuals with C9 and with ALS, excpet one).  Huh.  Considered chance, lab error, or whether there might be a 2nd mendelian gene.  Some evidence of that.  Or whether there's somatic instability.  Did indeed find somatic instability of the C9 repeat expansion.  Huh.

*Ahmad Al Khleifat* (King's college London)
Next-gen sequencing study of telomere length.  Describes structural variation (deletions, inversions, duplications, tandem repeats.)  Example is telomere.  Unfortunately I missed the info on the size of this study.  Telomere length comparison between ALS and healthy controls.  Age has an effect, gender does too, and so does case status (P=0.008).  Then survival analysis, longer telomeres survive longer (P=0.003).  Assessment of 9 loci previously associated with telomere length at 'genome-wide significance' in European populations.  Found rs6772228, rs8105767 are both associated with ALS.  Conclusion: longer telomeres associated with increased risk of ALS, this make a trade-off in risk. (But it seems like opposite effects have been [observed in mice](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4789589/).)

*Kevin Kenna* (Dublin) - about KIF5A as a novel ALS gene.
KIF5A identifed by LOF burden analysis.  2014 (n=363).  2018 (n=1,463), across many European (or european ancestry) cohorts.  Case-control gene burden analysis.  41,410 Control whole exomes.  This finds KIF5A with OR = 32 (9-135)!  And P=5.5x10^-7.  And other genes at 'exome-wide significance'.  (GB: jings, haven't we moved past that?)  Quick description of the bioinformatic work that goes into making this actually work well, including looking at qq plots, looking at possible confounders, validating KIF5A variants by sanger sequencing. Now table of patient variants.  Including exon skipping (splice) event.  Now: KIF5A also identified by GWAS.  8,000 cases and 36,000 controls.  20,416 cases vs 58,914 controls (McLaughlin, Nat Comm 2017).  Shows diagram of KIF5A, has 'motor domain', 'stalk', and 'tail'.  Most of these mutations occur in motor domain.  Involved in kinesin complex, which transports things along microtubules.  Now replication.  9,000 cases and 2,000 controls, replicated GWAS SNP though not rare mutation.  Summary: significant excess of KIF5A LOF mutations in ALS (high OR, rare atypical phenotype). (GB: though I don't believe the OR).  + interesting stuff about function.  Published in Neuron 21 March 2018, author list is 5 pages, it's this: <https://www.sciencedirect.com/science/article/pii/S089662731830148X>.  (I'm a bit worried about this (again).  They find a massive odds ratio in their large discovery cohort.  It will be enlarged through Winner's curse.  For that reason, usually people tkae the replication effect size.  But they don't replicate this in 11,000 samples (2).  Surely if the effect was that large it would replicate?)

(GB: there's a bunch of stuff going on here that coule be problematic.  First, the discovery phase rests on only 9 LOF variants in KIF5A: 6 in 1,138 cases and 3 in 19,494 controls.  That difference is highly statistically significant because of the large number of samples, but it's problematic because it's so easy for experimental or bioinformatic factors to introduce small numbers of counts like this, artifactually.  This is particularly so because the definition of LOF here includes 'splice sites' (splicing out of exons) that are predicted computationally based on DNA, RNA, and genetic variation databases.  So they might not be real!  The effect size seen in discovery is also absolutely massive (OR=32): it suffers massively from Winner's curse.  None of this is helped by references to 'exome-wide significance' which is not really helpful: it is a threshhold based on naive statistical arguments and not on actually pertinent information.)

*Lara Marrone* (Dresden) - Modelling FUS ALS using iPSC lines.
About iPSC-derived motor neurons.  I'm not following this (though I've previously seen a cool talk about this type of work from Justin Ichida).

*Mattia Perez* (Strasbourg)
About accumulation of 'exogenous recombinant FUS' in cortical neurons in mouse ALS models.  Protein aggregation is not randomly distributed across brain.  FUS is normally located in nucleus of cells, where it helps shuttle molecule between nucleus and cytoplasm.  Is FUS pathology spreading in the brain like a prion?  To answer: injected mice in the brain with recombinant FUS (poor mice again.  Haven't these people read the hitch-hikers guide?) But this is work in progress; aggregates were too sticky so results weren't reproducible.  So instead injected soluble FUS-GBP.  Sterotaxis injections in motor cortex and hippocampus (4-6month old mice).  Injected on day 0 and sacrificed after 3 and 30 days.  Used cell staining.  Staining implies cortical neurons do intake these proteins.  The FUS mutation does not modify entry into cortical neurons.  After 30 days, FUS-GFP immunoreactivity is weaker and confined to large aggreaget structures.  Same in mutant mice.  No obvious effect of the FUS mutants.  So what happens to endogenous FUS protein?   partial colocalisation.  Most obvious in hippocampus.  Might be technical issue, investigating this.  Conclusion: recombinant proteins can enter neurons, but not genotype dependent.  Do seem to be aggregate like structures that need to be identified an dmight recruit endogenous FUS.  Future: inject more mice.  Hmm.  Have also developed a mouse line in which can induce mutation in neurons in a labelled way.

Albert Rudolf (German Network for MND)
About safety and efficacy of Rasagline as an add-on therapy to riluzole.  A Randomised, double-blind, parallel-group, placebo-controlled trial.  Study partially supported by drug company.
Study rationale: previously, Rasagliline showed a significant, dose-dependent threapeutic effet on motor function and survuval in mice.  Largest extension of life (~20%) was seen in riluzole+rasagiline treatment.  Plots of suruvual.  Aparently improved after 6 and 12 months, but not by end of study (18 months).  In ALS-FRS-R, rasagiline has a significant effect.  COnclude there's a different of treatment effect in fast and slow progressors.  IN this study, the ALS/FRS shows a plateau in the beginning, predictive of further delince.  Rasagiline (1mg + riluzole) did not show effect on primary endpoint.  But post-hoc analysis showed effect at intermediate times.

---

More [tomorrow]({{ site.baseurl }}{% post_url 2018-06-21-ENCALS-Day-two-report %}).
